src/app/shared/equation.ts

```diff
@@ -215,6 +237,61 @@ export class Equation {
       return new Equation(this.RHS, this.LHS).solve(v);
     }
     switch (this.LHS.kind) {
+    case TypeDiscriminator.Logarithmize: {
+      const logarithmize = <Logarithmize>this.LHS;
+      if (!logarithmize.getAntilogarithm().contains(v)) {
+        const left = logarithmize.getBase();
+        const right = Equation.log(this.RHS, Equation.log(logarithmize.getAntilogarithm(), Equation.constantFromNumber(-1)));
+        return new Equation(left, right).solve(v);
+      }
+      return 'too complex';
+    }
+    case TypeDiscriminator.Antilogarithmize: {
+      const antilogarithmize = <Antilogarithmize>this.LHS;
+      if (!antilogarithmize.getLogarithm().contains(v)) {
+        const left = antilogarithmize.getBase();
+        const right = Equation.antilog(this.RHS, Equation.antilog(antilogarithmize.getLogarithm(), Equation.constantFromNumber(-1)));
+        return new Equation(left, right).solve(v);
+      }
+      return 'too complex';
+    }
```

Comment:
Instead of adding a whole new switch, put these cases in the existing switch below.